src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -205,8 +213,8 @@ public BufferedInputStream(InputStream in, int size) {
         if (size <= 0) {
             throw new IllegalArgumentException("Buffer size <= 0");
         }
-        buf = new byte[size];
-
+        buf = getClass() == BufferedInputStream.class ? EMPTY : new byte[size];
```

Comment:
Actually, you can merge this `getClass()` check with the one below for locks.

Reply:
Done